src/libsystemd/sd-bus/bus-objects.c

```diff
@@ -1690,10 +1726,14 @@ static int add_object_vtable_internal(
 
                 case _SD_BUS_VTABLE_METHOD: {
                         struct vtable_member *m;
+                        names_flags=1;
+                        names = strempty(v->x.method.names);
```

Comment:
you need to be more careful here. as mentioned earlier the structure is part of the ABI. we have now extended it, but old code will still use the old, shorter structure. This means in order to maintain ABI compat, you cannot rely on this field to exist, and you should only access it after validating that we are running with app code that is ready for it. Thankfully we already prepared for this case: the first entry in each vtable array is an ` _SD_BUS_VTABLE_START` entry that contains the actual size of the structure used (in the x.start.element_size field). This means you can check here whether the size stored there is larger than `offsetof(sd_bus_vtable, x.method.names)` and if so you know the field is available.

Please hence define a helper function or macro that dervices the `names` field of an sd_bus_vtable entry, that takes this into consideration and checks the element size first.